File: flow-designer/frontend/src/types/flow.ts
// ---- 后端数据结构 ----
export interface FlowNodeDef {
  id: string
  label: string
  position: { x: number; y: number }
  type?: string
}

export interface FlowEdgeDef {
  id: string
  source: string
  target: string
}

export interface FlowDefinition {
  nodes: FlowNodeDef[]
  edges: FlowEdgeDef[]
}

export interface Flow {
  id: number
  name: string
  flow_json: FlowDefinition
  create_time: string
  update_time: string
}

export interface FlowVersion {
  id: number
  flow_id: number
  version: number
  flow_json: FlowDefinition
  create_time: string
}

File: flow-designer/backend/app/models/flow.py
from sqlalchemy import Column, Integer, String, JSON, DateTime, ForeignKey
from sqlalchemy.sql import func
from app.db.database import Base


class Flow(Base):
    __tablename__ = "flows"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    flow_json = Column(JSON, nullable=False)
    create_time = Column(DateTime(timezone=True), server_default=func.now())
    update_time = Column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )


class FlowVersion(Base):
    __tablename__ = "flow_versions"

    id = Column(Integer, primary_key=True, index=True)
    flow_id = Column(Integer, ForeignKey("flows.id"), nullable=False, index=True)
    version = Column(Integer, nullable=False)
    flow_json = Column(JSON, nullable=False)
    create_time = Column(DateTime(timezone=True), server_default=func.now())

